libraries/bdp-access-proxy/bdp-access-mock/src/main/java/com/bosch/osmi/bdp/access/mock/BdpApiAccessMockImpl.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2015.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License Version 2.0 as published by the
+ * Free Software Foundation with classpath exception.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License version 2.0 for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program (please see the COPYING file); if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+package com.bosch.osmi.bdp.access.mock;
+
+import com.bosch.osmi.bdp.access.api.BdpApiAccess;
+import com.bosch.osmi.bdp.access.api.model.User;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.*;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+
+
+/**
+ * Allows an access to the Bdp APIs. An access has to be initialized with a username and a password for which
+ * a user object can be retrieved. The user object is the actual entry point into the whole object net retrieved
+ * from Bdp.
+ *
+ * @author muj1be
+ * @since 11/16/15.
+ */
+public class BdpApiAccessMockImpl implements BdpApiAccess {
+
+    public static final String MOCKDATA_CLASSPATH_LOCATION = "/mockdata.json";
```

Comment:
The mock reads the json file at runtime to provide the mock data and searches in the classpath for that file.